src/HDInsight/HDInsightOnAks.Autorest/README.md

```diff
@@ -0,0 +1,318 @@
+<!-- region Generated -->
+# Az.HdInsightOnAks
+This directory contains the PowerShell module for the HdInsightOnAks service.
+
+---
+## Status
+[![Az.HdInsightOnAks](https://img.shields.io/powershellgallery/v/Az.HdInsightOnAks.svg?style=flat-square&label=Az.HdInsightOnAks "Az.HdInsightOnAks")](https://www.powershellgallery.com/packages/Az.HdInsightOnAks/)
+
+## Info
+- Modifiable: yes
+- Generated: all
+- Committed: yes
+- Packaged: yes
+
+---
+## Detail
+This module was primarily generated via [AutoRest](https://github.com/Azure/autorest) using the [PowerShell](https://github.com/Azure/autorest.powershell) extension.
+
+## Module Requirements
+- [Az.Accounts module](https://www.powershellgallery.com/packages/Az.Accounts/), version 2.7.5 or greater
+
+## Authentication
+AutoRest does not generate authentication code for the module. Authentication is handled via Az.Accounts by altering the HTTP payload before it is sent.
+
+## Development
+For information on how to develop for `Az.HdInsightOnAks`, see [how-to.md](how-to.md).
+<!-- endregion -->
+
+---
+## Generation Requirements
+Use of the beta version of `autorest.powershell` generator requires the following:
+- [NodeJS LTS](https://nodejs.org) (10.15.x LTS preferred)
+  - **Note**: It *will not work* with Node < 10.x. Using 11.x builds may cause issues as they may introduce instability or breaking changes.
+> If you want an easy way to install and update Node, [NVS - Node Version Switcher](../nodejs/installing-via-nvs.md) or [NVM - Node Version Manager](../nodejs/installing-via-nvm.md) is recommended.
+- [AutoRest](https://aka.ms/autorest) v3 beta <br>`npm install -g autorest@autorest`<br>&nbsp;
+- PowerShell 6.0 or greater
+  - If you don't have it installed, you can use the cross-platform npm package <br>`npm install -g pwsh`<br>&nbsp;
+- .NET Core SDK 2.0 or greater
+  - If you don't have it installed, you can use the cross-platform npm package <br>`npm install -g dotnet-sdk-2.2`<br>&nbsp;
+
+## Run Generation
+In this directory, run AutoRest:
+> `autorest`
+
+---
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+### General settings
+> Values
+``` yaml
+branch: f09aacf4c6b63be416212cb182f6b31e8bc6d545
+require:
+  - $(this-folder)/../../readme.azure.noprofile.md
+  - $(repo)/specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/readme.md
+
+# input-file:
+#  - D:\source\azure-rest-api-specs\specification\hdinsight\resource-manager\Microsoft.HDInsight\HDInsightOnAks\preview\2023-06-01-preview\hdinsight.json
+inlining-threshold: 100
+```
+
+> Names
+``` yaml
+module-version: 0.1.0
+title: HdInsightOnAks
+subject-prefix: "$(service-name)"
+```
+
+> Exclude some properties in table view
+``` yaml
+# For a specific module, we could override this configuration by setting default-exclude-tableview-properties to false in readme.md of that module.
+default-exclude-tableview-properties: true
+```
+
+> Directives
+``` yaml
+identity-correction-for-post: true
```

Comment:
```suggestion
identity-correction-for-post: true

use-extension:
  "@autorest/powershell": "4.x"

```
I'm suggesting using higher version to generate module so that multiple features are available to customer, which also benefits to further development.

Reply:
updated